src/shared/Modal.tsx

```diff
@@ -0,0 +1,56 @@
+import { createSignal, Component, Show } from "solid-js";
+// Component Properties
+interface Props {
+  messageHeader: string;
+  message: string;
+  confirmText: string;
+  declineText: string;
+  onConfirm: () => void;
+}
+
+const Modal: Component<Props> = (props) => {
+  const [show, setShow] = createSignal(true);
+
+  const handleClose = () => {
+    setShow(false);
+  };
+  // Make sure you set an onConfirm function on the page or it will simply close the modal without performing a task!!
+  const handleConfirm = () => {
+    props.onConfirm();
+    setShow(false);
+  };
+
+  return (
+    <Show when={show()}>
+      <div class="fixed inset-x-0 bottom-0 px-4 pb-4 sm:inset-0 sm:flex sm:items-center sm:justify-center">
+        <div class="fixed inset-0 transition-opacity">
+          <div class="absolute inset-0 bg-gray-500 opacity-75" />
+        </div>
+        <div class="overflow-hidden rounded-lg bg-black px-4 pt-5 pb-4 shadow-xl transition-all sm:w-full sm:max-w-lg">
+          <div>
+            <div class="black mb-4 text-lg font-bold">
+              {props.messageHeader}
+            </div>
+            <div class="black mb-4 text-lg">{props.message}</div>
+            <div class="flex items-center justify-end">
+              <button
```

Comment:
Could there be a case where we may need more or less buttons than two? For example, a modal with information that doesn't need a confirm/deny action.

Reply:
Yep, I can also foresee situations where e.g. the button(s) should be disabled until a certain condition is met, and things of the sort.

I think the best interface for that will likely imply breaking the Modal component apart into `Modal.Header`, `Modal.Body` and `Modal.Footer` so we have fine-grained control of the contents in each through the `children` prop, but I think that can be left as a future PR if necessary.